addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -114,59 +159,21 @@ def _peppol_get_new_documents(self):
             for uuid, content in all_messages.items():
                 enc_key = content["enc_key"]
                 document_content = content["document"]
-                filename = content["filename"] or 'attachment' # default to attachment, which should not usually happen
+                filename = content["filename"] or f'peppol_document_{uuid}'  # default to peppol_document_{uuid}, which should not usually happen
                 partner_endpoint = content["accounting_supplier_party"]
                 decoded_document = edi_user._decrypt_data(document_content, enc_key)
 
-                journal_id = company.peppol_purchase_journal_id
-                # use the first purchase journal if the Peppol journal is not set up
-                # to create the move anyway
-                if not journal_id:
-                    journal_id = self.env['account.journal'].search([
-                        *self.env['account.journal']._check_company_domain(company),
-                        ('type', '=', 'purchase')
-                    ], limit=1)
-
                 attachment_vals = {
                     'name': f'{filename}.xml',
                     'raw': decoded_document,
                     'type': 'binary',
                     'mimetype': 'application/xml',
                 }
+                attachment = self.env['ir.attachment'].create(attachment_vals)
```

Comment:
Just minor suggestion, you don't have to follow it, but as you only use `attachment_vals` once now, it's a bit useless to define the variable to be used right below.